src/BeadingStrategy/CenterDeviationBeadingStrategy.cpp

```diff
@@ -6,97 +6,75 @@
+            }
         }
         else
         {
-            ret.left_over = thickness - bead_count * optimal_width_middle;
-        }
-
-        // last half minus middle:
-        for (coord_t bead_idx = (bead_count + 1) / 2; bead_idx < bead_count; bead_idx++)
-        {
-            ret.bead_widths.emplace_back(optimal_width_inner);
-            ret.toolpath_locations.emplace_back(thickness - (optimal_width_outer + ((bead_count - bead_idx) - 1) * optimal_width_inner) + optimal_width_inner / 2);
+            ret.left_over = thickness;
         }
+        return ret;
     }
-    else
-    {
-        ret.left_over = thickness;
-    }
-
-    return ret;
-}
-
-coord_t CenterDeviationBeadingStrategy::getOptimalThickness(coord_t bead_count) const
-{
-    return std::max(0LL, (bead_count - 1)) * optimal_width_inner + std::min(1LL, bead_count) * optimal_width_outer;
-}
 
-coord_t CenterDeviationBeadingStrategy::getTransitionThickness(coord_t lower_bead_count) const
-{
-    if (lower_bead_count % 2 == 0)
-    { // when we add the extra bead in the middle
-        return this->getOptimalThickness(lower_bead_count) + underfill_bound;
-    }
-    else
-    { // when we move away from the strategy which replaces two beads by a single one in the middle
-        return this->getOptimalThickness(lower_bead_count + 1) - overfill_bound;
-    }
-}
-
-coord_t CenterDeviationBeadingStrategy::getOptimalBeadCount(coord_t thickness) const
-{
-    coord_t thickness_left = thickness;
-    coord_t naive_count = std::min(1LL, ((thickness / 2 + optimal_width_outer / 2) / optimal_width_outer) * 2);
-    thickness_left -= naive_count * optimal_width_outer;
-    if (thickness_left >= (optimal_width_inner / 2))
+    coord_t CenterDeviationBeadingStrategy::getOptimalThickness(coord_t bead_count) const
     {
-        naive_count += ((thickness_left / 2 + optimal_width_inner / 2) / optimal_width_inner) * 2;
+        return bead_count * optimal_width;
     }
 
-    const coord_t overfill = this->getOptimalThickness(naive_count) - thickness;
-    if (overfill > overfill_bound)
-    {
-        return naive_count - 1;
-    }
-    else if (-overfill > underfill_bound)
+    coord_t CenterDeviationBeadingStrategy::getTransitionThickness(coord_t lower_bead_count) const
     {
-        return naive_count + 1;
+        if (lower_bead_count % 2 == 0)
+        { // when we add the extra bead in the middle
+            return lower_bead_count * optimal_width + underfill_bound;
+        }
+        else
+        { // when we move away from the strategy which replaces two beads by a single one in the middle
+            return (lower_bead_count + 1) * optimal_width - overfill_bound;
+        }
     }
-    else
+
+    coord_t CenterDeviationBeadingStrategy::getOptimalBeadCount(coord_t thickness) const
     {
-        return naive_count;
+        const coord_t naive_count = (thickness / 2 + optimal_width / 2) / optimal_width * 2;
```

Comment:
These expressions can be simplied to:

```c++
const coord_t optimal_thickness = thickness + optimal_width;
const coord_t naive_count = optimal_thickness  / optimal_width;
const coord_t overfill = optimal_thickness;
```
Maybe not do the last asignment of the `overfill` ;-) but replace the it further on in the codde with `optimal_thickness`

Reply:
It makes me wonder a bit if the logic is sound, eeeing how these equations simplify.